### PR TITLE
fix(translations): TAMOC-335: Delete translations for records with multiple text columns

### PR DESCRIPTION
**Pull request by tcodling**
### Changes

_Add a brief description of the changes in this PR to help give the reviewer context._

### Deploys

- [x] **Deploy to Tamanu Internal** 

### Tests

- [ ] **Run E2E Tests** 

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

---

> [!NOTE]
> Implements soft-deletion of translations for empty nested fields in the importer, updates translation creation logic, adds tests, and only renders survey helper text when detail exists.
> 
> - **Backend/Importer**:
>   - `translationHandler.generateTranslationsForData` now returns `{ createRecords, deleteClause }`, handles nested fields, sets `deletedAt` on created records, and builds delete clauses for empty fields.
>   - `importRows` consumes the new structure: conditionally bulk-creates translations (`updateOnDuplicate` includes `deletedAt`) and destroys translations via `deleteClause`.
> - **Tests**:
>   - Extend programs importer translation tests to cover nested option translations and soft-deletion when nested fields become empty.
> - **UI**:
>   - `SurveyQuestion.jsx`: render `helperText` only if `componentDetail` exists.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 19451cc31d3cf087aff24c4a94074b2c8fdef6ba. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by rohan-bes — CHANGES_REQUESTED** [review]

File: packages/central-server/app/admin/importer/translationHandler.js [hunk 1]
```diff
@@ -58,6 +58,13 @@ export function generateTranslationsForData(model, sheetName, values) {
         const stringId = `${REFERENCE_DATA_TRANSLATION_PREFIX}.${dataType}.${key}.${values.id}`;
         if (text) {
           translationData.push({ stringId, text, language: DEFAULT_LANGUAGE_CODE });
+        } else {
+          translationData.push({
+            stringId,
+            text: '',
```
[the remark this answers is not in the document]
Comment (rohan-bes) [reply packages/central-server/app/admin/importer/translationHandler.js]: It feels maybe a bit awkward to be create empty records for fields that never had a translation.

I think I'd lean towards changing this function to return to fields: `{ createRecords, deleteClause }`

Where `createRecords` is the current `translationData` and `deleteClause` is `{ language: DEFAULT_LANGUAGE_CODE, id: [<ids for deletion>] }`

Then the caller in `importRows` can perform `TranslatedString.bulkCreate(createRecords)`  and `TranslatedString.destroy({ where: deleteClause })` separately.

This also has the advantage of not generating unnecessary audit logs, as callling `destroy` on an already soft deleted record doesn't bump the `deletedAt` and trigger a new log (you might need to double check me on this, that's what AI says)

**Review by rohan-bes — CHANGES_REQUESTED** [review]
I think bot caught a bug, and a request to change the logic so we don't create unnecessary translations.

Also, this feels like a good corner case for some unit tests?

**Review by rohan-bes — APPROVED** [review]
Looks good! Thanks for making the changes
